yoyodyne/models/transducer.py

```diff
@@ -617,31 +617,16 @@ def forward(self, batch: batches.PaddedBatch) -> torch.Tensor:
         encoder_out = encoder_out[:, 1:, :]
         source_padded = batch.source.padded[:, 1:]
         source_mask = batch.source.mask[:, 1:]
-        # Converts features to n-hot encoding.
-        with torch.no_grad():
-            # Features are offset by features idx; we shift one to encode
-            # padding, replacing mask with first index.
-            features_padded = torch.where(
-                batch.features.mask,
-                0,
-                batch.features.padded - self.features_idx + 1,
-            )
-            # Features outside offset classified as unknown feature. The index
-            # is the end of the features vocabulary.
-            features_padded[features_padded < 0] = self.features_vocab_size + 1
-            # The size of the one-hot embeddings are the vocab size plus two
-            # for padding and unknown.
-            one_hot_features = nn.functional.one_hot(
-                features_padded, self.features_vocab_size + 2
-            )
-            # Truncates pad at idx 0 then sum one_hots for n_hot vectors.
-            n_hot_features = torch.sum(
-                one_hot_features[:, :, 1:], dim=1, keepdim=True
-            )
-        # B x 1 x n_feat -> B x seq_len x n_feat.
-        n_hot_features = n_hot_features.expand(-1, encoder_out.size(1), -1)
-        # Concatenates n-hot encoding onto each symbol.
-        encoder_out_feat = torch.cat((encoder_out, n_hot_features), dim=2)
+        # Prepares feature embeddings.
+        features_out = self.feature_embeddings(batch.features.padded)
+        features_out = torch.sum(features_out, dim=1)
+        denom = torch.sum(~batch.features.mask, dim=1, keepdim=True)
+        denom = denom.expand(-1, features_out.size(1))
+        features_out = features_out / denom
```

Comment:
Was this averaging in one of the publications for this model? If not, we should leave an option for the original behavior for reproducibility (I ask b/c it seems like features were handled differently before this PR).